packages/cubejs-materialize-driver/src/Stream.ts

```diff
@@ -0,0 +1,109 @@
+import { Readable } from 'stream';
+import { PoolClient } from 'pg';
+// eslint-disable-next-line import/no-extraneous-dependencies
+import { TypeId, TypeFormat } from 'pg-types';
+
+/**
+ * Thanks to Petros Angelatos
+ * https://gist.github.com/petrosagg/804e5f009dee1cb8af688654ba396258
+ * This class reads from a cursor in PostgreSQL
+ */
+export default class Stream extends Readable {
```

Comment:
This is a bit too low level and hard to maintain. Fortunately, modern Node has good support for Streams built out of async generators. See `AthenaDriver.stream` and `AthenaDriver.lazyRowIterator`. Unless there is a strong technical reason to roll our own Stream implementation.